beetsplug/fetchart.py

```diff
@@ -57,17 +73,26 @@ class Candidate:
     MATCH_FALLBACK = 1
 
     def __init__(
-        self, log, path=None, url=None, source="", match=None, size=None
+        self,
+        log,
+        source: "ArtSource",
```

Comment:
Note that I re-ordered arguments here to make `source` mandatory (which it effectively is, because it is used to index `SOURCE_NAMES`). If some third-party code added a new `ArtSource` and passed `source` as positional argument, this would be a breaking change. I guess that's extremely unlikely: I'm not sure there's such code, and it's also quite unlikely that somebody would pass `path` and `url` instead of using keyword arguments.